### PR TITLE
fix: editing of apphooked CMS pages without apphook landing page

### LINKED ISSUE #7765: [BUG] toolbar.obj is not set correctly in edit view when apphook landing page is a CMS page
### Summary

consider a an apphook that does not define a landing page (i.e. at location `""`), then this page is filled by the CMS with a standard `cms.Page`. but it seems like, this page is getting uneditable.

### Expected behaviour

It should be possible to edit the page that holds the apphook

### Actual behaviour

When one creates a new draft, the toolbar is filled with the _published_ content object, not with the _draft_ content object. therefore all the links in the toolbar are not working because one is redirected to the _structure_ mode of the published content object (that is of course not editable anymore), and not to the real one. The reason is that we loose the `object_id` when the `view_func` is called in `cms.views.render_object_endpoint`

https://github.com/django-cms/django-cms/blob/c20b042edd717d46c585583010aef2c4ca395c5d/cms/views.py#L282-L294

for a page with an apphook that does not define a view at location `""`, this `view_func` resolves to the `cms.views.details` view. and `cms.views.details` uses the standard `page.get_content_obj` that always resolves to the _published_ page content

https://github.com/django-cms/django-cms/blob/c20b042edd717d46c585583010aef2c4ca395c5d/cms/views.py#L214

therefore the `toolbar` gets the wrong `obj` and generates the wrong urls

### Environment

* Python version: 3.11.4
* Django version: 3.2.23
* django CMS version: 4.1.0

### PR DESCRIPTION
**Pull request by Chilipp**
### Summary

Fixes #7765

### Proposed changes in this pull request
this implements a small patch and a test to make sure, that CMS pages that hold an apphook without a urlpattern at the `""`-location still have the correct pagecontent object at the edit url

### Documentation checklist

* [ ] I have updated CHANGELOG.txt if appropriate
* [ ] I have updated the release notes document if appropriate, with:
    * [ ] general notes
    * [ ] bug-fixes
    * [ ] improvements/new features
    * [ ] backwards-incompatible changes
    * [ ] required upgrade steps
    * [ ] names of contributors
* [ ] I have updated other documentation
* [ ] I have added my name to the AUTHORS file
* [ ] This PR's documentation has been approved by Daniele Procida

### CODE REVIEW

**Comment by Chilipp** [comment]
@fsbraun I implemented the changes that we discussed in #7765. As this is my first PR to django-cms, I am not sure, where to edit the `CHANGELOG.txt`. seems like there is no section for `4.1.1` yet. shall I create one?

**Comment by fsbraun** [comment]
@Chilipp No need to change the Changelog. It will be automatically generated by the release process.

**Comment by Chilipp** [comment]
@fsbraun this PR should work now (at least the tests pass locally)

**Review by fsbraun — APPROVED** [review]
LGTM! Nice! Thank you!

**Comment by fsbraun** [comment]
@Chilipp I sneaked in a little unrelated change in the docs to get this out. Hope that was ok.

**Comment by Chilipp** [comment]
Sure, it's your code :wink:

File: docs/introduction/01-install.rst [hunk 1]
```diff
@@ -273,10 +274,12 @@ We suggest to use pip-compile to freeze your requirements as, for example, discu
 Spin up your Django development server (Step 3)
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Now you are ready to spin up Django's development server by running:
+Now you are ready to spin up Django's development server by first changing directory into 
+the projcet folder and then spinning up the development server:
```
Comment (fsbraun) [inline docs/introduction/01-install.rst:278]:
```suggestion
the project folder and then spinning up the development server:
```

**Pull request merged**